core/templates/dev/head/pages/exploration_editor/ParamMetadataObjectFactory.js

```diff
@@ -0,0 +1,60 @@
+// Copyright 2018 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Factory for creating new frontend instances of ParamMetadata
+ * domain objects.
+ */
+
+oppia.factory('ParamMetadataObjectFactory', [function() {
+  /**
+   * @private @constructor
+   * Provide metadata when a parameter is changed by a GET or SET action
+   * @param {String} action - set or get
+   * @param {String} paramName - parameter's name
+   * @param {String} source - location where the parameter was defined
+   * e.g. answer, content, feedback or param_changes(changing value of param)
+   * @param {String} sourceInd - index for multiple actions with the same source
```

Comment:
Could you elaborate on this a bit more, please? I'm not sure I understand how this index is actually calculated. Do you mean something like "index of this parameter in a set of parameter changes"?

Reply:
Yes, I think I was trying to say "within the same source" i.e multiple param_changes or multiple 
"feedback"  parameter changes